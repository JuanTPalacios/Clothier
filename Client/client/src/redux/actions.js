const actions = {
  getItems : (items) => ({
    type: 'getItems',
    payload: items
  }),
};

export default actions;