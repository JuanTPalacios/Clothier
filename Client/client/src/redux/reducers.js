import { combineReducers } from 'redux';

const initialState = {
  items: [] ,
  setAuthenticated: false
}

const items = (state = initialState, action) => {
  switch (action.type) {
    case 'getItems':
    return {items: [...state.items, ...action.payload]};
  default:
    return state;
  }
};

// Combining both reducers
const reducers = combineReducers({
  items,
});

export default reducers;