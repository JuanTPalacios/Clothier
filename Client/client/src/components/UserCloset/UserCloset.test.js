import {render, screen} from '@testing-library/react';
import UserCloset from './UserCloset';
import { Provider } from 'react-redux';
import reducers from '../../redux/reducers';
import { createStore } from 'redux';
import ReactRouter from 'react-router'
let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ &&
  window.__REDUX_DEVTOOLS_EXTENSION__());

describe('Register Page', () => {
  test('should match the snapshot', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ userId: 1 });
    const { container } = render(<Provider store={store}><UserCloset /></Provider>);
    expect(container.firstChild).toMatchSnapshot();
  })
  test('should have correct title', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ userId: 1 });
    render(<Provider store={store}><UserCloset /></Provider>);
    screen.getByText(/'s Closet/);

  })
})