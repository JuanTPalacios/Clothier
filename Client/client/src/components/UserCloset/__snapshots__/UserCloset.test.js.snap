// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page should match the snapshot 1`] = `
<div>
  
  <div
    class="body"
  >
    <h1
      class="title is-0 mt-5 ml-5 mb-0"
    >
      's Closet
    </h1>
    <div
      class="mt-2 ml-5"
    >
      <button
        class="button is-success is-rounded"
      >
        Follow
      </button>
    </div>
    <div
      class="categories tabs"
    >
      <ul>
        <li
          class="is-active"
        >
           
          <a>
            All
          </a>
           
        </li>
      </ul>
    </div>
    <div
      class="dashboardItems tile is-ancestor"
    >
      <div
        class="tile is-3 is-vertical pt-2"
      />
      <div
        class="tile is-3 is-vertical"
      />
      <div
        class="tile is-3 is-vertical pt-3"
      />
      <div
        class="tile is-3 is-vertical"
      />
    </div>
  </div>
</div>
`;
